src/Listener/UpdateCastorListener.php

```diff
@@ -0,0 +1,159 @@
+
+use Castor\Console\Application;
+use Castor\PlatformUtil;
+use JoliCode\PhpOsHelper\OsHelper;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Process\Process;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class UpdateCastorListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly CacheItemPoolInterface&CacheInterface $cache,
+        private readonly HttpClientInterface $httpClient,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {
+    }
+
+    public function checkUpdate(ConsoleCommandEvent $event): void
+    {
+        if (class_exists(\RepackedApplication::class)) {
+            return;
+        }
+        if (PlatformUtil::getEnv('DISABLE_VERSION_CHECK')) {
+            return;
+        }
+
+        $command = $event->getCommand();
+        if (!$command) {
+            return;
+        }
+        if ('_complete' === $command->getName()) {
+            return;
+        }
+        $input = $event->getInput();
+        if ($input->hasOption('format') && 'json' === $input->getOption('format')) {
+            return;
+        }
+
+        $this->displayUpdateWarningIfNeeded($input, $event->getOutput());
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            // Must be before the command is executed, because we have to check
+            // for many command options
+            ConsoleEvents::COMMAND => 'checkUpdate',
+        ];
+    }
+
+    private function displayUpdateWarningIfNeeded(InputInterface $input, OutputInterface $output): void
+    {
+        $item = $this->cache->getItem('castor-releases');
+
+        if ($item->isHit()) {
+            $latestVersion = $item->get();
+        } else {
+            $latestVersion = null;
+            $item->expiresAfter(60 * 60 * 24);
+
+            try {
+                $latestVersion = $this
+                    ->httpClient
+                    ->request('GET', 'https://api.github.com/repos/jolicode/castor/releases/latest', [
+                        'timeout' => 1,
+                    ])
+                    ->toArray()
+                ;
+            } catch (ExceptionInterface) {
+                $this->logger->info('Failed to fetch latest Castor version from GitHub.');
+
+                $item->expiresAfter(60 * 10);
+
+                return;
+            }
+
+            $this->cache->save($item->set($latestVersion));
+        }
```

Comment:
So you reverted to the previous behaviour displaying update message everytime? :smile:

Reply:
Yes! It makes more sens actually, isn't ? You were right! Because right now, you have one change per day to see the upgrade message... 😬 

But now, a lot of things are actually cache...

Reply:
Yes, I think it's better like that :+1: